hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -264,6 +265,163 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     }
   }
 
+  @Override
+  public boolean viewExists(TableIdentifier identifier) {
+    return HiveCatalogUtil.isTableWithTypeExists(clients, identifier, TableType.VIRTUAL_VIEW);
+  }
+
+  @Override
+  public boolean dropView(TableIdentifier identifier) {
+    if (!isValidIdentifier(identifier)) {
+      return false;
+    }
+
+    try {
+      String database = identifier.namespace().level(0);
+      String viewName = identifier.name();
+      Table table = clients.run(client -> client.getTable(database, viewName));
+      HiveCatalogUtil.validateTableIsIcebergView(table, fullTableName(name, identifier));
+      clients.run(
+          client -> {
+            client.dropTable(database, viewName);
+            return null;
+          });
+      LOG.info("Dropped View: {}", identifier);
+      return true;
+
+    } catch (NoSuchViewException | NoSuchObjectException e) {
+      LOG.info("Skipping drop, View does not exist: {}", identifier, e);
+      return false;
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop " + identifier, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to dropView", e);
+    }
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    Preconditions.checkArgument(
+        isValidateNamespace(namespace), "Missing database in namespace: %s", namespace);
+
+    try {
+      return listSpecificTables(namespace, TableType.VIRTUAL_VIEW.name());
+    } catch (UnknownDBException e) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all views under namespace " + namespace, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listViews", e);
+    }
+  }
+
+  private List<TableIdentifier> listAllTables(Namespace namespace)
+      throws TException, InterruptedException {
+    String database = namespace.level(0);
+    List<String> tableNames = clients.run(client -> client.getAllTables(database));
+    return tableNames.stream()
+        .map(t -> TableIdentifier.of(namespace, t))
+        .collect(Collectors.toList());
+  }
+
+  private List<TableIdentifier> listSpecificTables(Namespace namespace, String tableType)
+      throws TException, InterruptedException {
+    String database = namespace.level(0);
+    List<String> tableNames =
+        clients.run(client -> client.getTables(database, "*", TableType.valueOf(tableType)));
+    // Each TableData is a big object.Retrieving it with a big number can put pressure on memory.
+    // To be safe, it is being retired in batches.
+    int offset = 0;
+    int limit = 100;
+    List<TableIdentifier> filteredTableIdentifiers = Lists.newArrayList();
+    while (offset < tableNames.size()) {
```

Comment:
Maybe we could use some tooling, like https://www.baeldung.com/java-stream-batch-processing?